control-plane/connect-inject/endpoints_controller_ent_test.go

```diff
@@ -90,37 +91,41 @@ func TestReconcileCreateEndpointWithNamespaces(t *testing.T) {
 			k8sObjects: func() []runtime.Object {
 				pod1 := createPodWithNamespace("pod1", test.SourceKubeNS, "1.2.3.4", true, true)
 				pod2 := createPodWithNamespace("pod2", test.SourceKubeNS, "2.2.3.4", true, true)
-				endpointWithTwoAddresses := &corev1.Endpoints{
+				endpointslice := &v1.EndpointSlice{
 					ObjectMeta: metav1.ObjectMeta{
-						Name:      "service-created",
-						Namespace: test.SourceKubeNS,
+						Name:      "service-created-123456",
```

Comment:
Most of the test changes are simply creating an EndpointSlice object instead of Endpoints objects.
Careful attention will also show that the Name of the EndpointSlice object is no longer synonymous with the Kube Service it is serving. This uses the full DNS subdomain name (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names).

The controller will then use either the OwnerReference (which is the actual Kube Service), or will parse this dns name in the event that there is no OwnerReference (such as in the case of delete).